Commonalities/documentation/API-design-guidelines.md

```diff
@@ -1021,3 +1027,214 @@ Finally, this part describes the OAuth security applied to the API. This spec is
+    },
+    "uePort": 5060
+  },
+  "subscriptionExpireTime": "2023-03-31T00:00:00.000Z"
+}
+```
+
+response:
+
+```json
+201 Created
+
+{
+  "notificationUrl": "https://application-server.com/v0",
+  "notificationAuthToken": "c8974e592c2fa383d4a3960714",
+  "subscriptionDetail": {
+    "ueId": {
+      "ipv4Addr": "192.168.0.1"
+    },
+    "uePort": 5060
+  },
+  "subscriptionExpireTime": "2023-03-31T00:00:00.000Z",
+  "subscriptionId": "456g899g",
+  "eventType": "ROAMING_STATUS",
+  "startedAt": "2023-03-17T16:02:41.314Z"
+}
+```
+
+
+
+
+### 12.2 Event notification
+
+The notification endpoint is used by the API server to notify the API consumer that an event occured.
+
+Note: The notification is the message posted on listener side. We describe the notification message in the CAMARA OAS but it could confusing because this endpoint should be implemented on the business API consumer side. This notice should be explicited mentioned in all CAMARA API documentation featuring notification.
+
+Only Operation POST is provided for `notifications` and the expected response code is `204`.
+
+For consistence between CAMARA API a uniform `notifications` model must be used:
+
+| name | type | attribute description | cardinality |
+| ----- |	-----  |	 -----  |  -----  | 
+| subscriptionId | string | subscription identifier - could be valued for Resource-based subscription | optional |
+| eventType | string | type of event as defined in each CAMARA API. The event type are defined in UPPER_SNAKE_CASE| mandatory |
+| eventTime | string - datetime | date time when the event was notified by the issuer- valued in the request. This is not the date when the event occurred | mandatory |
+| eventDetail | object | A detailed event structure depending on the eventType | mandatory |
+
+
+Specific eventType "SUBSCRIPTION_ENDS" is defined to inform listener about subscrition termination. It is used when the subscription expire time (required by the requester) has been reached or if the API server has to stop sending notification prematurely. For this specific event, the 'eventDetail' must feature 'terminationReason' attribute.
+
+_Examples_
+
+Example for Roaming status - Request:
+
+```json
+curl -X 'POST' \
+  'https://application-server.com/v0/notifications' \
+  -H 'accept: application/json' \
+  -H 'Content-Type: application/json' \
+  -d
+{
+  "subscriptionId": "456g899g",
+  "eventType": "ROAMING_STATUS",
+  "eventTime": "2023-01-19T13:18:23.682Z",
+  "eventDetail": {
+    "ueId": {
+      "ipv4Addr": "192.168.0.1"
+      },
+    "uePort": 5060,
+    "roaming": true,
+    "countryCode": 208,
+    "countryName": "FR"
+  }
+}
+```
+
+response:
+
+```json
+204 No Content
+```
+
+Example for subscription termination - Request:
+
+```json
+curl -X 'POST' \
+  'https://application-server.com/v0/notifications' \
+  -H 'accept: application/json' \
+  -H 'Content-Type: application/json' \
+  -d
+{
+  "subscriptionId": "456g899g",
+  "eventType": "SUBSCRIPTION_ENDS",
```

Comment:
Let's discuss it. I can understand that it could be optional for this event type but in this case it means that we will shift eventDetail to optional and that will be the case for all kind of event.